### ISSUE TITLE
[feature]: coverage for CSharp applications

### ISSUE THREAD

**Issue by shivamsouravjha** [keploy/keploy#2052]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Summary

* Just like node, python, Java and Go there should also be coverage for Csharp applications.

### Why should this be worked on?

* Csharp dependencies will be  run with coverage 

### Repository

keploy

**Comment by Swpn0neel** [comment]
Hey @yaten2302, thanks for your interest 😊

But, kindly complete the PRs of the issues that are currently assigned to you, after that we can assign this issue to you!

**Comment by debayangg** [comment]
@Swpn0neel @shivamsouravjha would like to work on this issue

**Comment by yaten2302** [comment]
@Swpn0neel , @shivamsouravjha sir, may I collaborate with @debayangg on this issue? My previous issues assigned to me are complete 👍

**Comment by Sonichigo** [comment]
@debayangg any updates ?

**Comment by debayangg** [comment]
@Sonichigo I have just gone through the basics of how the coverage might be implemented. I haven't started working on a PR. Should I work on this issue?

**Comment by Sonichigo** [comment]
You can pickup and try :) ! Lmk if you face any issue regading anything.

**Comment by yaten2302** [comment]
@Swpn0neel @Sonichigo , if no one is working on this, then may I start solving this issue?

**Comment by Swpn0neel** [comment]
Go ahead @yaten2302 !!

**Comment by yaten2302** [comment]
@Swpn0neel @Sonichigo @shivamsouravjha , I was researching about generating C# coverage reports, what I've understood is that:
1. We've to look for `.xml` or `.json` files probably in the `coverage/` or `bin/` dir.
2. Then we've to check that if the dotnet-coverage doesn't exists, then we've to install it for further use.
3. We then have to merge all the files together using the dotnet-coverage library.
4. Then we can create one single file i.e. `target/keploy-e2e...` and append all the generated items in it.

Could you please guide that am I correct?

**Comment by shivamsouravjha** [comment]
@yaten2302 seems like the correct approach but instead of manually merging them we can leverage any library, example incase of java we use java -jar merge. But we must ensure that merge is correct, we don't want to merge a coverage file which is old and now not present in new coverage

Also please see wether we can get the output in cobertura format.

**Comment by yaten2302** [comment]
@shivamsouravjha , yes, we'll not merge them manually, but with the help of `dotnet-coverage` lib. 
Can we use this command, is this correct?
https://learn.microsoft.com/en-us/dotnet/core/additional-tools/dotnet-coverage#dotnet-coverage-merge
